src/SayMore/Model/ArchivingHelper.cs

```diff
@@ -397,14 +396,13 @@ private static string GetAnalysisLanguageIdentifier(Project saymoreProject)
 			return analysisLanguage;
 		}
 
-		private static string GetRole(string personId, ArchivingActorCollection actors, System.Collections.Generic.IEnumerable<string> allParticipantsWithRoles)
+		private static string GetRole(string personId, ArchivingActorCollection actors, ContributionCollection contributions)
 		{
-			return (from p in allParticipantsWithRoles
-				where p.StartsWith(personId + " (")
-				let openParen = p.IndexOf("(", StringComparison.InvariantCulture)
-				let closeParen = p.IndexOf(")", StringComparison.InvariantCulture)
-				select p.Substring(openParen + 1, closeParen - openParen - 1))
-				.FirstOrDefault(role => !PersonHasRole(personId, actors, role));
+			if (contributions == null)
```

Comment:
This could be made more compact (and IMO, more readable) as follows:
return contributions?.Where(c => c.ContributorName == personId).Select(c => c.Role).FirstOrDefault()?.Name;